src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -42,23 +42,30 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var cancellationToken = context.CancellationToken;
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start,
-                token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
-            if (identifier == default)
+            LocalFunctionStatementSyntax localFunction = default;
```

Comment:
i would extract this all out into a helper that returns a LocalFunction upon success.  like TryGetSelectedLocalFunction.  All the assigning to default and having to do all these checks just feels baroque.

Reply:
Sure, thought about that and wanted to wait until I find another place where it could be used but you're right. Extracting it even with one use-case has readability benefits.

Reply:
Any preference on where such method should live? In the type itself or do we see it as something that might be useful e.g. in `SyntaxTreeExtensions`?

Reply:
In this type is fine.